### PR TITLE
refactor/CLI - Changed assets folder structure

### PR DESCRIPTION
**Pull request by rafapaulin**
- Base path must now be set in a (optional) config.i18n.json file on the same folder as the translation files
  - Config can be set in a component-level
  - basePath: String: Sets the namespace of the translation file. Defaults to 'base'
  - loadMode: ['lazy|'eager']: Sets how the translation file should be loaded. Defaults to 'eager'
- Assets folder structure changed to support both eager and lazy loaded translation files
  - All the eager-loaded translation now sits on assets/i18n/{locale}/base.i18n.json
  - All the lazy-loaded translation now sits on assets/i18n/{locale}/{config.basePath}.i18n.json

### CODE REVIEW

File: projects/smarti18n/cli/commandBuild.js [hunk 1]
```diff
@@ -21,23 +21,24 @@ module.exports = class BuildCommand {
...
-		const list = fs.readdirSync(buildPath);
-		const sep = this.separateFolders(buildPath, list);
-		sep.folders.forEach(folder => this.build(folder));
-		if (sep.i18n)
-			this.addDataFromDir(sep.i18n);
+		const LIST = fs.readdirSync(buildPath);
+		const SEP = this.separateFolders(buildPath, LIST);
+		SEP.folders.forEach(folder => this.build(folder));
+		if (SEP.i18n)
+			this.addDataFromDir(SEP.i18n);
```
Comment (vertexportus) [inline projects/smarti18n/cli/commandBuild.js]: why change these to uppercase? I understand changing the src path but these I don't see why...

File: projects/smarti18n/cli/commandBuild.js [hunk 2]
```diff
@@ -96,20 +109,45 @@ module.exports = class BuildCommand {
...
 		}
+
+		writeData = this.parseDotNotation(basePath, writeData);
+
 		// write the rest
-		Object.keys(readData).forEach(k => k !== '_basePath' ? writeData[k] = readData[k] : null);
+		Object.keys(readData).forEach(k => {
+			const TRANSLATION = basePath.split('.').reduce((a, b) => a[b], writeData);
+
+			TRANSLATION[k] = readData[k];
```
Comment (vertexportus) [inline projects/smarti18n/cli/commandBuild.js]: same here with uppercase...

File: projects/smarti18n/cli/commandBuild.js [hunk 3]
```diff
@@ -119,9 +157,13 @@ module.exports = class BuildCommand {
 	 */
 	saveLocales(saveBasePath) {
 		Object.keys(this.locales).forEach(locale => {
-			const finalPath = path.join(saveBasePath, 'assets', 'i18n', `${locale}.i18n.json`);
-			fs.outputJsonSync(finalPath, this.locales[locale]);
-			console.log(chalk.green(`${finalPath} written successfully`));
+			const OBJ = this.locales[locale];
+			Object.keys(OBJ).forEach(key => {
+				const FINAL_PATH = path.join(saveBasePath, 'assets', 'i18n', locale, `${key}.i18n.json`);
```
Comment (vertexportus) [inline projects/smarti18n/cli/commandBuild.js]: same here with obj and final_path

**Review by vertexportus — CHANGES_REQUESTED** [review]
you've changed several `const` variables into uppercase CONSTANTS, there's a very big difference between the two - i'd prefer keeping global constants as uppercase, and local `const` variables as lower camel case

File: projects/smarti18n/cli/commandBuild.js (hunk 1 above)
Comment (rafapaulin) [reply projects/smarti18n/cli/commandBuild.js]: I just like to keep constants as uppercase lol, my syntax highlights recognize them this way...

But I can revert that

Comment (vertexportus) [reply projects/smarti18n/cli/commandBuild.js]: yes but I usually prefer to use all uppercase to signalize class-wide or global-wide constants, not scoped `const variables`

File: projects/smarti18n/cli/commandBuild.js [hunk 4]
```diff
@@ -133,16 +175,34 @@ module.exports = class BuildCommand {
...
 		const angularConfigPath = path.join(process.cwd(), 'angular.json');
-		if (!fs.existsSync(angularConfigPath))
+		if (!fs.existsSync(angularConfigPath)) {
+			console.log(chalk.red('error: Angular config not found!'));
 			throw new Error('Angular config not found');
+		}
+
 		const angularConfig = require(angularConfigPath);
-		project = project || angularConfig.defaultProject;
+		project = project || angularConfig._DEFAULTPROJECT;
```
Comment (vertexportus) [inline projects/smarti18n/cli/commandBuild.js]: this is an angular-cli default, should not change the name of the property from `defaultProject`

**Review by vertexportus — CHANGES_REQUESTED** [review]

File: projects/smarti18n/cli/commandBuild.js (hunk 4 above)
Comment (rafapaulin) [reply projects/smarti18n/cli/commandBuild.js]: fixed

**Review by vertexportus — APPROVED** [review]

**Pull request merged**